Explorer/Assets/Scripts/ECS/StreamableLoading/Textures/TexturesCache.cs

```diff
@@ -0,0 +1,84 @@
+﻿using Cysharp.Threading.Tasks;
+using DCL.Optimization.PerformanceBudgeting;
+using DCL.Profiling;
+using ECS.StreamableLoading.Cache;
+using ECS.StreamableLoading.Common.Components;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.Pool;
+using Utility;
+using Utility.Multithreading;
+using Utility.PriorityQueue;
+
+namespace ECS.StreamableLoading.Textures
+{
+    public class TexturesCache : IStreamableCache<Texture2D, GetTextureIntention>
+    {
+        private readonly Dictionary<GetTextureIntention, Texture2D> cache;
+        private readonly SimplePriorityQueue<GetTextureIntention, long> unloadQueue = new ();
+
+        public IDictionary<string, UniTaskCompletionSource<StreamableLoadingResult<Texture2D>?>> OngoingRequests { get; }
+
+        public IDictionary<string, StreamableLoadingResult<Texture2D>> IrrecoverableFailures { get; }
+
+        private bool disposed { get; set; }
+
+        public TexturesCache()
+        {
+            cache = new Dictionary<GetTextureIntention, Texture2D>(this);
+
+            IrrecoverableFailures = DictionaryPool<string, StreamableLoadingResult<Texture2D>>.Get();
```

Comment:
We don't need pools as this cache is shared (singleton). I made the same changes for AssetBundleCache in `main`